# Pigpen Graph Test

To reproduce run

```bash
lein test
```

Produces this error:

```
ERROR in (test-roundtrip) (nippy.clj:508)
Uncaught exception, not in assertion.
expected: nil
  actual: java.lang.Exception: Thaw failed: Encrypted data or wrong compressor?
 at taoensso.nippy$thaw$ex__4715.doInvoke (nippy.clj:508)
    clojure.lang.RestFn.invoke (RestFn.java:423)
    taoensso.nippy$thaw$try_thaw_data__4719.invoke (nippy.clj:524)
    taoensso.nippy$thaw.doInvoke (nippy.clj:549)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    nippy_aot_record_test/fn (nippy_aot_record_test.clj:8)
    clojure.test$test_var$fn__7187.invoke (test.clj:704)
    clojure.test$test_var.invoke (test.clj:704)
    clojure.test$test_vars$fn__7209$fn__7214.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars$fn__7209.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars.invoke (test.clj:718)
    clojure.test$test_all_vars.invoke (test.clj:728)
    clojure.test$test_ns.invoke (test.clj:747)
    clojure.core$map$fn__4245.invoke (core.clj:2559)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.boundedLength (RT.java:1654)
    clojure.lang.RestFn.applyTo (RestFn.java:130)
    clojure.core$apply.invoke (core.clj:626)
    clojure.test$run_tests.doInvoke (test.clj:762)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:624)
    user$eval85$fn__140$fn__171.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140$fn__141.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140.invoke (form-init3972934316697246797.clj:1)
    user$eval85.invoke (form-init3972934316697246797.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:6703)
    clojure.lang.Compiler.eval (Compiler.java:6693)
    clojure.lang.Compiler.load (Compiler.java:7130)
    clojure.lang.Compiler.loadFile (Compiler.java:7086)
    clojure.main$load_script.invoke (main.clj:274)
    clojure.main$init_opt.invoke (main.clj:279)
    clojure.main$initialize.invoke (main.clj:307)
    clojure.main$null_opt.invoke (main.clj:342)
    clojure.main$main.doInvoke (main.clj:420)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
Caused by: java.lang.Exception: Thaw failed against type-id: 80
 at taoensso.nippy$thaw_from_in.invoke (nippy.clj:469)
    taoensso.nippy$thaw_from_in_BANG_.doInvoke (nippy.clj:475)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    taoensso.nippy$thaw$try_thaw_data__4719.invoke (nippy.clj:518)
    taoensso.nippy$thaw.doInvoke (nippy.clj:549)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    nippy_aot_record_test/fn (nippy_aot_record_test.clj:8)
    clojure.test$test_var$fn__7187.invoke (test.clj:704)
    clojure.test$test_var.invoke (test.clj:704)
    clojure.test$test_vars$fn__7209$fn__7214.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars$fn__7209.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars.invoke (test.clj:718)
    clojure.test$test_all_vars.invoke (test.clj:728)
    clojure.test$test_ns.invoke (test.clj:747)
    clojure.core$map$fn__4245.invoke (core.clj:2559)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.boundedLength (RT.java:1654)
    clojure.lang.RestFn.applyTo (RestFn.java:130)
    clojure.core$apply.invoke (core.clj:626)
    clojure.test$run_tests.doInvoke (test.clj:762)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:624)
    user$eval85$fn__140$fn__171.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140$fn__141.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140.invoke (form-init3972934316697246797.clj:1)
    user$eval85.invoke (form-init3972934316697246797.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:6703)
    clojure.lang.Compiler.eval (Compiler.java:6693)
    clojure.lang.Compiler.load (Compiler.java:7130)
    clojure.lang.Compiler.loadFile (Compiler.java:7086)
    clojure.main$load_script.invoke (main.clj:274)
    clojure.main$init_opt.invoke (main.clj:279)
    clojure.main$initialize.invoke (main.clj:307)
    clojure.main$null_opt.invoke (main.clj:342)
    clojure.main$main.doInvoke (main.clj:420)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
Caused by: java.lang.ClassNotFoundException: nippy_aot_record_test.foo
 at java.net.URLClassLoader$1.run (URLClassLoader.java:366)
    java.net.URLClassLoader$1.run (URLClassLoader.java:355)
    java.security.AccessController.doPrivileged (AccessController.java:-2)
    java.net.URLClassLoader.findClass (URLClassLoader.java:354)
    java.lang.ClassLoader.loadClass (ClassLoader.java:425)
    sun.misc.Launcher$AppClassLoader.loadClass (Launcher.java:308)
    java.lang.ClassLoader.loadClass (ClassLoader.java:358)
    java.lang.Class.forName0 (Class.java:-2)
    java.lang.Class.forName (Class.java:190)
    taoensso.nippy$thaw_from_in.invoke (nippy.clj:441)
    taoensso.nippy$thaw_from_in_BANG_.doInvoke (nippy.clj:475)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    taoensso.nippy$thaw$try_thaw_data__4719.invoke (nippy.clj:518)
    taoensso.nippy$thaw.doInvoke (nippy.clj:549)
    clojure.lang.RestFn.invoke (RestFn.java:410)
    nippy_aot_record_test/fn (nippy_aot_record_test.clj:8)
    clojure.test$test_var$fn__7187.invoke (test.clj:704)
    clojure.test$test_var.invoke (test.clj:704)
    clojure.test$test_vars$fn__7209$fn__7214.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars$fn__7209.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars.invoke (test.clj:718)
    clojure.test$test_all_vars.invoke (test.clj:728)
    clojure.test$test_ns.invoke (test.clj:747)
    clojure.core$map$fn__4245.invoke (core.clj:2559)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.boundedLength (RT.java:1654)
    clojure.lang.RestFn.applyTo (RestFn.java:130)
    clojure.core$apply.invoke (core.clj:626)
    clojure.test$run_tests.doInvoke (test.clj:762)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:624)
    user$eval85$fn__140$fn__171.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140$fn__141.invoke (form-init3972934316697246797.clj:1)
    user$eval85$fn__140.invoke (form-init3972934316697246797.clj:1)
    user$eval85.invoke (form-init3972934316697246797.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:6703)
    clojure.lang.Compiler.eval (Compiler.java:6693)
    clojure.lang.Compiler.load (Compiler.java:7130)
    clojure.lang.Compiler.loadFile (Compiler.java:7086)
    clojure.main$load_script.invoke (main.clj:274)
    clojure.main$init_opt.invoke (main.clj:279)
    clojure.main$initialize.invoke (main.clj:307)
    clojure.main$null_opt.invoke (main.clj:342)
    clojure.main$main.doInvoke (main.clj:420)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)

Ran 1 tests containing 1 assertions.
0 failures, 1 errors.
Tests failed.
```
